python/pyarrow/table.pyx

```diff
@@ -296,6 +296,56 @@ cdef class RecordBatch:
         return result
 
 
+cdef class RecordBatchList:
+    '''
+    A utility class used for conversions with RecordBatch lists
+    '''
+
+    @staticmethod
+    def to_pandas(batches):
+        cdef:
+            vector[vector[shared_ptr[CArray]]] c_array_vec
+            shared_ptr[CColumn] c_col
+            PyObject* np_arr
+            Array arr
+            Schema schema
+
+        import pandas as pd
+
+        schema = batches[0].schema
+        K = batches[0].num_columns
+
+        # TODO - check schemas are equal
+
+        # make a vector of ArrayVectors to store column chunks
+        for i in range(K):
+            c_array_vec.push_back(vector[shared_ptr[CArray]]())
+
+        # copy each batch into a chunk
+        for batch in batches:
+            for i in range(K):
+                arr = batch[i]
+                c_array_vec[i].push_back(arr.sp_array)
```

Comment:
If you instead loop over column index, then loop over batches, you can avoid this `c_array_vec` business and construct the Column directly. See Table.to_pandas